src/Mollie/WC/Helper/ApplePayDataObjectHttp.php

```diff
@@ -0,0 +1,470 @@
+    {
+        $this->resetErrors();
+        if (!$this->hasRequiredFieldsValues(
+            $post,
+            self::VALIDATION_REQUIRED_FIELDS
+        )
+        ) {
+            return;
+        }
+        $this->assignDataObjectValues($post);
+    }
+
+    /**
+     * Set the object with the data relevant to ApplePay on update shipping contact
+     * Required data depends on callerPage
+     * @param array $post
+     */
+    public function updateContactData(array $post)
+    {
+        $this->resetErrors();
+        $requiredFields = self::UPDATE_CONTACT_SINGLE_PROD_REQUIRED_FIELDS;
+        if (array_key_exists('callerPage', $post)
+            && $post['callerPage'] == 'cart'
+        ) {
+            $requiredFields = self::UPDATE_CONTACT_CART_REQUIRED_FIELDS;
+        }
+        if (!$this->hasRequiredFieldsValues(
+            $post,
+            $requiredFields
+        )
+        ) {
+            return;
+        }
+        $this->assignDataObjectValues($post);
+        $this->simplifiedContact = $this->simplifiedAddress(
+            filter_var_array(
+                $post['simplifiedContact'],
+                FILTER_SANITIZE_STRING
+            )
+        );
+    }
+
+    /**
+     * Set the object with the data relevant to ApplePay on update shipping method
+     * Required data depends on callerPage
+     *
+     * @param array $post
+     */
+    public function updateMethodData(array $post)
+    {
+        $this->resetErrors();
+        $requiredFields = self::UPDATE_METHOD_SINGLE_PROD_REQUIRED_FIELDS;
+
+        if (array_key_exists('callerPage', $post)
+            && $post['callerPage'] == 'cart'
+        ) {
+            $requiredFields = self::UPDATE_METHOD_CART_REQUIRED_FIELDS;
+        }
+        if (!$this->hasRequiredFieldsValues(
+            $post,
+            $requiredFields
+        )
+        ) {
+            return;
+        }
+        $this->assignDataObjectValues($post);
+
+        $this->simplifiedContact = $this->simplifiedAddress(
+            filter_var_array(
+                $post['simplifiedContact'],
+                FILTER_SANITIZE_STRING
+            )
+        );
+        $this->shippingMethod = filter_var_array(
+            $post['shippingMethod'],
+            FILTER_SANITIZE_STRING
+        );
+    }
+
+    /**
+     * Set the object with the data relevant to ApplePay on authorized order
+     * Required data depends on callerPage
+     *
+     * @param array $post
+     * @param       $callerPage
+     */
+    public function orderData(array $post, $callerPage)
+    {
+        $this->resetErrors();
+        $requiredFields = self::CREATE_ORDER_SINGLE_PROD_REQUIRED_FIELDS;
+        if ($callerPage == 'cart') {
+            $requiredFields = self::CREATE_ORDER_CART_REQUIRED_FIELDS;
+        }
+        if (!$this->hasRequiredFieldsValues(
+            $post,
+            $requiredFields
+        )
+        ) {
+            return;
+        }
```

Comment:
This instruction it's difficult to read, can be better formatted? Even if you need to create a variable.